src/lightning/pytorch/trainer/connectors/logger_connector/result.py

```diff
@@ -362,6 +362,23 @@ def log(
         rank_zero_only: bool = False,
     ) -> None:
         """See :meth:`~lightning.pytorch.core.module.LightningModule.log`"""
+        # For torchmetric.Metric, if sync_dist is True,
+        # set `sync_on_compute` and `dist_sync_on_step` properties
+        # based on the on_step and on_epoch arguments.
+        if isinstance(value, Metric) and sync_dist:
+            if on_step or on_epoch:
+                value.sync_on_compute = sync_dist
+                value.dist_sync_on_step = on_step
+
+            # All 3 (on_step, on_epoch and sync_dist) should not be set to True
+            if on_step and on_epoch and sync_dist:
+                raise MisconfigurationException(
```

Comment:
The exception should be raised before the `if` above is checked, otherwise you are running it when you will fail anyways